lib/custom/src/MShop/Service/Provider/Payment/Mollie.php

```diff
@@ -47,4 +47,250 @@ protected function getValue( $key, $default = null )
+		}
+
+		return $order;
+	}
+	
+	/**
+	 * Tries to get an authorization or captures the money immediately for the given order if capturing the money
+	 * separately isn't supported or not configured by the shop owner.
+	 *
+	 * @param \Aimeos\MShop\Order\Item\Iface $order Order invoice object
+	 * @param array $params Request parameter if available
+	 * @return \Aimeos\MShop\Common\Item\Helper\Form\Standard Form object with URL, action and parameters to redirect to
+	 * 	(e.g. to an external server of the payment provider or to a local success page)
+	 */
+	protected function processOrder( \Aimeos\MShop\Order\Item\Iface $order, array $params = [] )
+	{
+		$parts = \Aimeos\MShop\Order\Item\Base\Base::PARTS_SERVICE
+			| \Aimeos\MShop\Order\Item\Base\Base::PARTS_PRODUCT
+			| \Aimeos\MShop\Order\Item\Base\Base::PARTS_ADDRESS;
+
+		$base = $this->getOrderBase( $order->getBaseId(), $parts );
+		$data = $this->getData( $base, $order->getId(), $params );
+		$urls = $this->getPaymentUrls();
+
+		try
+		{
+			$provider = $this->getProvider();
+
+			if( $this->getValue( 'authorize', false ) && $provider->supportsAuthorize() )
+			{
+				$response = $provider->authorize( $data )->send();
+				$status = \Aimeos\MShop\Order\Item\Base::PAY_AUTHORIZED;
+			}
+			elseif( $this->getValue( 'apiType', '' ) === 'order' )
+			{
+				$response = $provider->createOrder( $data )->setItems( $this->getItems( $base, $params ) )->send();
+				$status = \Aimeos\MShop\Order\Item\Base::PAY_AUTHORIZED;
+			}
+			else
+			{
+				$response = $provider->purchase( $data )->send();
+				$status = \Aimeos\MShop\Order\Item\Base::PAY_RECEIVED;
+			}
+
+			if( $response->isSuccessful() )
+			{
+				$this->saveTransationRef( $base, $response->getTransactionReference() );
+
+				$order->setPaymentStatus( $status );
+				$this->saveOrder( $order );
+			}
+			elseif( $response->isRedirect() )
+			{
+				if( ( $ref = $response->getTransactionReference() ) != null ) {
+					$this->saveTransationRef( $base, $ref );
+				}
+
+				return $this->getRedirectForm( $response );
+			}
+			else
+			{
+				$order->setPaymentStatus( \Aimeos\MShop\Order\Item\Base::PAY_REFUSED );
+				$this->saveOrder( $order );
+
+				throw new \Aimeos\MShop\Service\Exception( $response->getMessage() );
+			}
+		}
+		catch( \Exception $e )
+		{
+			throw new \Aimeos\MShop\Service\Exception( $e->getMessage() );
+		}
+
+		return new \Aimeos\MShop\Common\Item\Helper\Form\Standard( $urls['returnUrl'], 'POST', [] );
+	}
+	
+	/**
+	 * Returns an Omnipay credit card object, except for phone numbers as Mollie requests a strict number
+	 * format which we can not provice in all cases.
+	 *
+	 * @param \Aimeos\MShop\Order\Item\Base\Iface $base Order base object with addresses and services
+	 * @param array $params POST parameters passed to the provider
+	 * @return \Omnipay\Common\CreditCard Credit card object
+	 */
+	protected function getCardDetails( \Aimeos\MShop\Order\Item\Base\Iface $base, array $params )
```

Comment:
Why did you overwrite that method and removed some values?

Reply:
Mollie API / Klarna is very strict on the phone number format. I'm not sure if an automatic transformation is a good idea when it comes to the payment process. So I removed the phone numbers altogether, Klarna will ask for them at a later stage if necessary.

Addiontally, GDPR makes it difficult to pass this information, IMHO.

Reply:
From the GDPR point of view, the shipping address and the e-mail should be removed as well

Reply:
Right, but they are necessary for the Orders API. Phone number is not. So I think a shop provider could argue with Article 6 here.